### PR TITLE
Task 10

### PR DESCRIPTION
**Pull request by kirillnaumenko**
Added task 01 with unit tests

### CODE REVIEW

File: src/test/java/UnitTests/JsonParserTests/JsonParserReadTests.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import parser.Parser;
+import shop.Cart;
+import java.io.File;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+
+public class JsonParserReadTests {
+
+    @ParameterizedTest
+    @ValueSource(strings = { "sample-cart.json" })
+    void readFromFileTest(String fileName) {
```
Comment (KBugrimCS) [inline src/test/java/UnitTests/JsonParserTests/JsonParserReadTests.java]: please do not forget to add access modifiers for each method

File: src/test/java/UnitTests/JsonParserTests/JsonParserReadTests.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        Parser parser = new JsonParser();
+        Cart cart;
+
+        cart = parser.readFromFile(new File("./src/test/java/UnitTests/TestData/" + fileName));
+        cart.showItems();
+
+        Assert.assertNotNull(cart);
```
Comment (KBugrimCS) [inline src/test/java/UnitTests/JsonParserTests/JsonParserReadTests.java]: Please add error message as a second param here, and it's good approach to add such messages for all assertions

File: src/test/java/UnitTests/JsonParserTests/JsonParserReadTests.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        });
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = { "wrong-datatype-cart.json" })
+    void readFromWrongDataTypeTest(String fileName) {
+        Assertions.assertThrows(NumberFormatException.class, () -> {
+            new JsonParser().readFromFile(new File("./src/test/java/UnitTests/TestData/" + fileName)).showItems();
+        });
+    }
```
Comment (KBugrimCS) [inline src/test/java/UnitTests/JsonParserTests/JsonParserReadTests.java]: please take a look these both tests and readFromWrongExtensionTypeTest absolutely the same except a params value. No need to copy paste these tests, junit allows you to unite them in 1 test with a collection of params

File: src/test/java/UnitTests/JsonParserTests/JsonParserWriteTests.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    @BeforeEach
+    void SetUp(){
+        cart = new Cart("sample-cart" + LocalDateTime.now().getSecond());
+    }
+
+    @AfterEach
+    void CleanUp(){
+        File cartFile = new File("src/main/resources/" + cart.getCartName()+ ".json");
+        if (cartFile.delete()) {
+            System.out.println("Deleted the file: " + cartFile.getName());
```
Comment (KBugrimCS) [inline src/test/java/UnitTests/JsonParserTests/JsonParserWriteTests.java]: let's use some logger, for example logback or log4j2 instead of using sout

**Review by KBugrimCS — CHANGES_REQUESTED** [review]
Good, but need to fix a couple moments:
1. Do not forget about access modifiers
2. Always add some error messages to assertions
3. Using some logger is a good approach instaed of using println method

File: src/test/java/UnitTests/JsonParserTests/JsonParserReadTests.java (hunk 3 above)
Comment (kirillnaumenko) [reply src/test/java/UnitTests/JsonParserTests/JsonParserReadTests.java]: not quite the same. Each test catch different exception NullPointerException and NumberFormatException. I didnt find a way to be able to catch different exceptions in the same test. Only like that

**Review by KBugrimCS — APPROVED** [review]

**Pull request merged**